src/app/redmine.service.spec.ts

```diff
@@ -1,12 +1,81 @@
 import { TestBed } from '@angular/core/testing';
+import {
+  HttpClientTestingModule,
+  HttpTestingController
+} from '@angular/common/http/testing';
 
 import { RedmineService } from './redmine.service';
+import { WeekLog } from './models/time-entries';
+
+
+const mockUserResponse = {
+  user: {
+    api_key: 'xxxxxxxxxxxxxxxx',
+    created_on: '2019-07-01T11:13:34Z',
+    firstname: 'John',
+    id: 11,
+    last_login_on: '2019-07-25T19:59:10Z',
+    lastname: 'Doe',
+    login: 'john.doe@fragaria.cz',
+    mail: 'john.doe@fragaria.cz',
+
+  }
+};
+
+const mockListEntryResponse = {
+  time_entries: [],
+  total_count: 0,
+  offset: 0,
+  limit: 0
+};
 
 describe('RedmineService', () => {
-  beforeEach(() => TestBed.configureTestingModule({}));
+  let httpTestingController: HttpTestingController;
+  let service: RedmineService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [RedmineService],
+      imports: [HttpClientTestingModule]
+    });
+
+    httpTestingController = TestBed.get(HttpTestingController);
+    service = TestBed.get(RedmineService);
+  });
+
+  afterEach(() => {
+    httpTestingController.verify();
+  });
 
   it('should be created', () => {
-    const service: RedmineService = TestBed.get(RedmineService);
     expect(service).toBeTruthy();
   });
+
+  it('should authenticate', () => {
+    service.authenticate('username', 'password')
+      .subscribe(data => {
+        expect(data).toBe(mockUserResponse.user);
+        expect((service as any).currentUser).toBe(mockUserResponse.user);
+      });
+
+    const req = httpTestingController.expectOne('/api/users/current.json');
+    expect(req.request.method).toEqual('GET');
+    req.flush(mockUserResponse);
```

Comment:
Do not use the same object as mocked data and in expectation statements.

Reply:
Is it better to check the type of incoming data in the expectation?